### ISSUE TITLE
Querying the last checking result in kalu-dbus

### ISSUE THREAD

**Issue by yan12125** [jjk-jacky/kalu#46]
I often ssh back to my machine. For Ubuntu machines, motd contains the information for how many packages can be updated. I'd like to implement a similar feature in Arch Linux via kalu. Does these information provided in kalu-dbus?

**Comment by jjk-jacky** [comment]
No, kalu-dbus is basically internal stuff, there's no reason for it
to be used by anything but kalu itself. More importantly, it's only to
deal with kalu's updater.

There's no way to do what you want per-se, as in get the last check
results from kalu. However, you could have a "separate use" of kalu,
that is run it in a script (e.g. from cron job) and capture its output.

If you ran e.g. `kalu -a` from a terminal (or script) it will print its
results on stdout instead of using notifications, you can then
parse/use that as you wish.

**Comment by yan12125** [comment]
Thanks for explaining the usage of kalu-dbus. How about another approach: in each periodic check of kalu-gui, write the check result to a temporary file. With that my scripts can read from it and parse it.

I know ```kalu -a```, but it always download the latest ```*.db``` files, which takes time and not suitable for motd.

**Comment by jjk-jacky** [comment]
I don't really like the idea of writing a tmp file that's not really
needed (by kalu) for anything, no.

Here's another idea: for some time now I've been meaning to do some
changes related to CLI usage, but never really found the time to work
on it. Specifically, I'd like for kalu (in cli mode) to be able to
keep & re-use an old tmp folder.

Let me explain a bit: in order to run its checks, kalu creates a tmp
folder where sync dbs are copied, and synced if needed. This is what
allows to run checks w/out root permissions, or messing up the system
dbs.
Nowadays the "GUI kalu" will not remove this folder when running, only
when exiting. This allows to re-use said folder/tmp dbs when running
checks again, thus not needing to sync them if there were no changes
since last checks (even though system dbs might be older/needing sync).
Saves a bit a time/bandwidth on occasion.
(This is why if you have kalu running right now, and checks ran at least
once, you'll have a kalu-XXXXX folder in your /tmp)

I've been meaning to add a few options to kalu (for use in cli
mode, though they're not restricted to it):
- one so it doesn't remove its tmp folder on exit (so it can be re-used)
- one to specify which folder to use as tmp folder, for the same
  reason. (Obviously, specifying a folder that's not properly set up
  will not get anything, but that's to be expected.)

Now, what I could also add is an option so kalu doesn't sync dbs. This
one would only make sense/be usable when in cli mode, and would allow
you to ask kalu to just "re-calculate" the results using a given tmp
folder, w/out syncing dbs -- so it should be pretty quick.

So, you/your script could get the folder name from /tmp, and
basically run `kalu -a --with-tmp-folder=/tmp/kalu-* --keep-tmp-folder
--dont-sync-dbs` to get results right away.

Would that work for you?

**Comment by yan12125** [comment]
These options sounds great! Much thanks for the work on this project.

**Comment by jjk-jacky** [comment]
Alright, so after thinking about this some more, I don't think that
last option (--no-sync) is a good idea after all.

The first two should definitely happen, as they'll allow "cli kalu" to
benefit from the same feature as "gui kalu" - so we don't have to
re-download dbs if that can be avoided/has already been done before.

As you mentioned your issue with re-syncing dbs, I thought this option
(--no-sync) would basically provide a "re-show last notifications"-like
feature for cli kalu; I was wrong.

Thing is, this only applies to ALPM-related checks, and that's only 2
out of 5. kalu would still require to do queries for news & (Watched)
AUR.

So I'm going back to the original idea: for what you want, I think e.g.
a cronjob that periodically runs kalu and saves its output to a file or
something, is the way to go.
(And with the other 2 options, you'd be able to have this kalu use the
same tmp folder/dbs as your gui kalu.)

**Comment by yan12125** [comment]
OK. Not noticing that. Thanks for the explanation.
